src/providers/local/exit-pool.provider.ts

```diff
@@ -218,22 +313,41 @@ const provider = (props: Props) => {
   // any existing input.
   watch(hasFetchedPoolsForSor, () => {
     debounceQueryExit.value();
+    debounceGetSingleAssetMax.value();
   });
 
-  watch(isSingleAssetExit, newVal => {
+  watch(isSingleAssetExit, _isSingleAssetExit => {
+    bptIn.value = '';
     queryError.value = '';
-    exitPoolService.setExitHandler(newVal);
+    exitPoolService.setExitHandler(_isSingleAssetExit);
+    debounceGetSingleAssetMax.value();
   });
 
+  watch(
+    () => singleAmountOut.address,
+    async () => {
+      await Promise.all([
+        debounceGetSingleAssetMax.value,
+        debounceQueryExit.value,
```

Comment:
Should call the functions, so the new values are calculated
```suggestion
        debounceGetSingleAssetMax.value(),
        debounceQueryExit.value(),
```